### PR TITLE
Bug 1480513: Add telemetry for search shortcut

### PR DESCRIPTION
**Pull request by ncloudioj**
This documents the telemetry part for [search shortcuts](https://bugzilla.mozilla.org/show_bug.cgi?id=1479806) in Top Sites of Activity Stream.

@fmarier Could you take a look at this please? The data review request was attached at [here](https://bugzilla.mozilla.org/show_bug.cgi?id=1480513#c2)

### CODE REVIEW

File: docs/v2-system-addon/data_events.md [hunk 1]
```diff
@@ -90,8 +90,53 @@ A user event ping includes some basic metadata (tab id, addon version, etc.) as
   "source": "TOP_SITES",
   "action_position": 2,
   "value": {
-    "card_type": "pinned",
-    "icon_type": ["screenshot_with_icon" | "screenshot" | "tippytop" | "rich_icon" | "no_image"]
+    "card_type": ["pinned" | "search"],
+    "icon_type": ["screenshot_with_icon" | "screenshot" | "tippytop" | "rich_icon" | "no_image"],
+    "search_vendor": ["google" | "amazon" | "baidu" | "yandex"]
```
Comment (k88hudson) [inline docs/v2-system-addon/data_events.md]: This needs to be added to `Blocking a site` and pinning as well I think. Or would pinning be covered by adding a search shortcut?

Comment (ncloudioj) [reply docs/v2-system-addon/data_events.md]: Yep, this will be added to those two events.

**Comment by ncloudioj** [comment]
r? @piatra, I figure you're the best reviewer for this patch, since you added both "card_type" and "searchTopSite" :)

Note that the "ADD_SEARCH_SHORTCUT" and "EDIT_SEARCH_SHORTCUT" events would be implemented separately by Bug 1480507, I've added the document for them here so that we only need one data review for the whole feature.

File: docs/v2-system-addon/data_events.md [hunk 2]
```diff
@@ -136,8 +181,9 @@ A user event ping includes some basic metadata (tab id, addon version, etc.) as
   "source": "TOP_SITES",
   "action_position": 2,
   "value": {
-    "card_type": "pinned",
-    "icon_type": ["screenshot_with_icon" | "screenshot" | "tippytop" | "rich_icon" | "no_image"]
+    "card_type": ["pinned" | "search"],
+    "icon_type": ["screenshot_with_icon" | "screenshot" | "tippytop" | "rich_icon" | "no_image"],
+    "search_vendor": ["google" | "amazon"]
```
[the remark this answers is not in the document]
Comment (ncloudioj) [reply docs/v2-system-addon/data_events.md]: I am not sure about the definite vendor list atm, I'll add a note for this field in data_dictionary.md

File: docs/v2-system-addon/data_dictionary.md [hunk 3]
```diff
@@ -183,7 +183,8 @@ Schema definitions/validations that can be used for tests can be found in `syste
 | `action` | [Required] Either `activity_stream_event`, `activity_stream_session`, or `activity_stream_performance`. | :one:
 | `addon_version` | [Required] Firefox build ID, i.e. `Services.appinfo.appBuildID`. | :one:
 | `client_id` | [Required] An identifier for this client. | :one:
-| `card_type` | [Optional] ("bookmark", "pocket", "trending", "pinned") | :one:
+| `card_type` | [Optional] ("bookmark", "pocket", "trending", "pinned", "search") | :one:
+| `search_vendor` | [Optional] the vendor of the search shortcut, one of ("google", "amazon", "wikipedia", "duckduckgo", "bing", etc.). This field only exists when `card_type = "search"` | :one:
```
Comment (fmarier) [inline docs/v2-system-addon/data_dictionary.md]: I might be misreading the code in the previous file, but wouldn't `this.props.link.hostname` return `google.com`, `amazon.com` or `wikipedia.org`?

Comment (ncloudioj) [reply docs/v2-system-addon/data_dictionary.md]: I believe `hostname` is already shortened by [shortURL](https://github.com/mozilla/activity-stream/blob/0cf21c41e96d81cfa143b7da96a9959f8e4c4abf/lib/ShortURL.jsm#L38)

**Comment by ncloudioj** [comment]
Data review+ at https://bugzilla.mozilla.org/show_bug.cgi?id=1480513#c5

**Pull request merged**